### PR TITLE
feat: migrate from CRA4 to Vite, Storybook to v8, jest to Vitest

### PR DESCRIPTION
**Pull request by karrui**
This PR moves the frontend toolchain from CRA4 to Vite. 
Generally everything seems to be working fine, including decryption workers.

Requires a bunch of changes that has to be performed all at the same time:
- [x] Jest -> Vitest
- [x] Storybook v6 -> v8 (less painful than wrangling Storybook v6 to work with vite)

# TODO:
- [x] move back dd-chunk pipeline to vite build
- [ ] Test on GSIB

### CODE REVIEW

**Comment by karrui** [comment]
* **#7308** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7308?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7305** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7305?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7292** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7292?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7285** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7285?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#7279** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7279?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7278** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7278?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7277** <a href="https://app.graphite.dev/github/pr/opengovsg/FormSG/7277?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by karrui** [comment]
@KenLSM see chromatic useful. will ensure the migration has 0 frontend changes!!!

**Comment by karrui** [comment]
@justynoh this PR officially has no regressions hehehehe

**Comment by karrui** [comment]
@justynoh added datadog chunking back to the pipeline. Can push to staging to try?

File: frontend/package.json [hunk 1]
```diff
@@ -102,20 +106,18 @@
...
+    "gen:theme-typings": "chakra-cli tokens src/theme/index.ts --out node_modules/@chakra-ui/styled-system/dist/declarations/src/theming.types.d.ts || true",
     "clean:emotion-types": "rimraf node_modules/@emotion/core/types",
     "postinstall": "patch-package && npm run gen:theme-typings && npm run clean:emotion-types && npm --prefix ../shared install",
-    "start": "env-cmd -f .buildtime-env craco --openssl-legacy-provider start",
-    "build": "npm run build:dd-chunk && cross-env CI=false BUILD_PATH='../dist/frontend' INLINE_RUNTIME_CHUNK=false SKIP_PREFLIGHT_CHECK=true craco build && cd ../ && cp -rf frontend/static/js dist/frontend/static",
-    "build:dd-chunk": "webpack --config webpack.dd.config.js",
-    "test": "cross-env SKIP_PREFLIGHT_CHECK=true craco test --silent",
+    "start": "vite",
+    "build": "vite build",
+    "serve": "vite preview",
```
Comment (karrui) [inline frontend/package.json]: can remove don't add confusion

File: frontend/package.json [hunk 2]
```diff
@@ -102,20 +106,18 @@
...
+    "test": "cross-env SKIP_PREFLIGHT_CHECK=true vitest",
     "eject": "react-scripts eject",
```
Comment (karrui) [inline frontend/package.json]: no need this anymore

File: frontend/package.json [hunk 3]
```diff
@@ -102,20 +106,18 @@
...
-    "storybook": "NODE_OPTIONS=--openssl-legacy-provider start-storybook -p 6006",
-    "build-storybook": "NODE_OPTIONS=--openssl-legacy-provider build-storybook",
+    "storybook": "NODE_OPTIONS=--openssl-legacy-provider storybook dev -p 6006",
```
Comment (karrui) [inline frontend/package.json]: no need node options anymore

Comment (KenLSM) [reply frontend/package.json]: 2856457124b1c2e7c0d27697e82661f9df658916

File: frontend/package.json (hunk 2 above)
Comment (KenLSM) [reply frontend/package.json]: 2856457124b1c2e7c0d27697e82661f9df658916

File: frontend/package.json (hunk 1 above)
Comment (KenLSM) [reply frontend/package.json]: 2856457124b1c2e7c0d27697e82661f9df658916

**Review by KenLSM — APPROVED** [review]

File: shared/package.json [hunk 4]
```diff
@@ -8,6 +8,8 @@
     "jszip": "^3.7.1",
     "libphonenumber-js": "^1.10.48",
     "lodash": "^4.17.21",
+    "tweetnacl": "^1.0.3",
+    "tweetnacl-util": "^0.15.1",
```
Comment (KenLSM) [inline shared/package.json:12]: @kevin9foong btw we forgot to add these two packages in to `/shared` 💀. 

Finally solved final piece of why chromatic publishing [fails](https://github.com/opengovsg/FormSG/actions/runs/10843484420/job/30090747001) when using vite/rollup. `webpack` is a little looser in resolving imports thus we didn't notice it while `rollup/pnpm` are much stricter.

**Comment by karrui** [comment]
this one got some regressions on chromatic, do you want to check?

**Pull request merged**